### PR TITLE
People search results after back button

### PR DESCRIPTION
**Pull request by antoniavonto**
Instead of retaining the people search results, this is calling the search function with the retained parameters to simulate retaining the search results.

Issue: https://github.com/gordon-cs/gordon-360-ui/issues/1851

### CODE REVIEW

File: src/views/PeopleSearch/components/SearchFieldList/index.tsx [hunk 1]
```diff
@@ -173,8 +173,8 @@ const SearchFieldList = ({ onSearch }: Props) => {
    * Whether the user can search for the current params.
    * This prevents a search with empty params, which freezes the client by trying to render thousands of results
    */
-  const canSearch = useMemo(() => {
-    const { includeStudent, includeFacStaff, includeAlumni, ...criteria } = searchParams;
+  const canSearch = (params: PeopleSearchQuery) => {
+    const { includeStudent, includeFacStaff, includeAlumni, ...criteria } = params;
```
Comment (amos-cha) [inline src/views/PeopleSearch/components/SearchFieldList/index.tsx:177]: why was this changed?

Comment (antoniavonto) [reply src/views/PeopleSearch/components/SearchFieldList/index.tsx:177]: because we are calling the search function with the params taken from the URL before the setSearchParams gets to update. So the search function will call canSearch, but this time it should check if it can search with the params of the URL instead of the searchParams that was not updated yet

**Comment by EjPlatzer** [comment]
Just checking, did we decide to do this instead of retaining search results because retaining search results was too difficult? I have a PR, #1992, that successfully retains search results.

**Comment by antoniavonto** [comment]
> Just checking, did we decide to do this instead of retaining search results because retaining search results was too difficult? I have a PR, #1992, that successfully retains search results.

This was to substitute retain search params, but if there is a solution already I can close this branch and the issue in the board.

**Comment by antoniavonto** [comment]
After talking to @russtuck  and @jsenning , we decided to merge this for now and later on think about an improvement for efficiency. Instead of researching every time to actually retain the search results.
